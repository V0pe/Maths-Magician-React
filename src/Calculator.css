body {
    margin:0;
    padding: 0;
    box-sizing: border-box;
}

main {
    display: grid;
    grid-template-columns: auto auto auto auto;
    background-color: grey;
    color: black;
    text-align: center; 
    position: absolute;
    right: 5rem;
    top: 10rem;
    gap:0.1rem;
    width: 50%;
    min-height: 50vh;

}

.item {
    padding:2rem;
    background-color: antiquewhite;
    font-size: 1.5rem;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
}  

.result {
    grid-column: 1/5;
    background-color: grey;
    display: flex;
    justify-content: flex-end;
    color: white;
}

.no-0 {
    grid-column: 1/3;
}

.divide,
.multiply,
.substract,
.add,
.equal {
    background-color: orange;
}