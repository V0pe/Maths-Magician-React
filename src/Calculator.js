// eslint-disable-next-line react/prefer-stateless-function
import React from 'react';
import './Calculator.css';

class Calculator extends React.PureComponent {
  render() {
    return (
      <main>
        <div className="item result"> 0 </div>
        <div className="item reset"> AC </div>
        <div className="item order"> +/- </div>
        <div className="item percent"> % </div>
        <div className="item divide"> / </div>
        <div className="item no-7"> 7 </div>
        <div className="item no-8"> 8  </div>
        <div className="item no-9"> 9 </div>
        <div className="item multiply"> X </div>
        <div className="item no-4"> 4 </div>
        <div className="item no-5"> 5 </div>
        <div className="item no-6"> 6 </div>
        <div className="item substract"> -  </div>
        <div className="item no-1"> 1 </div>
        <div className="item no-2"> 2 </div>
        <div className="item no-3"> 3 </div>
        <div className="item add"> + </div>
        <div className="item no-0"> 0 </div>
        <div className="item dot"> . </div>
        <div className="item equal"> =  </div>

      </main>
    );
  }
}

export default Calculator;
