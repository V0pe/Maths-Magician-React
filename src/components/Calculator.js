// eslint-disable-next-line react/prefer-stateless-function
import React from 'react';
import './Calculator.css';
import Calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };

    this.eventClick = this.eventClick.bind(this);
  }

  eventClick(e) {
    this.setState((state) => ({
      ...state,
      ...Calculate(state, e.target.name),
    }));
  }

  render() {
    const { total, next, operation } = this.state;

    return (
      <main>
        <div id="result" className="item result">
          {total}
          {operation}
          {next}
        </div>
        <button type="button" className="item reset" name="AC" onClick={this.eventClick}> AC </button>
        <button type="button" className="item order" name="+/-" onClick={this.eventClick}> +/- </button>
        <button type="button" className="item percent" name="%" onClick={this.eventClick}> % </button>
        <button type="button" className="item divide" name="รท" onClick={this.eventClick}> / </button>
        <button type="button" className="item no-7" name="7" onClick={this.eventClick}> 7 </button>
        <button type="button" className="item no-8" name="8" onClick={this.eventClick}> 8  </button>
        <button type="button" className="item no-9" name="9" onClick={this.eventClick}> 9 </button>
        <button type="button" className="item multiply" name="x" onClick={this.eventClick}> X </button>
        <button type="button" className="item no-4" name="4" onClick={this.eventClick}> 4 </button>
        <button type="button" className="item no-5" name="5" onClick={this.eventClick}> 5 </button>
        <button type="button" className="item no-6" name="6" onClick={this.eventClick}> 6 </button>
        <button type="button" className="item substract" name="-" onClick={this.eventClick}> - </button>
        <button type="button" className="item no-1" name="1" onClick={this.eventClick}> 1 </button>
        <button type="button" className="item no-2" name="2" onClick={this.eventClick}> 2 </button>
        <button type="button" className="item no-3" name="3" onClick={this.eventClick}> 3 </button>
        <button type="button" className="item add" name="+" onClick={this.eventClick}> + </button>
        <button type="button" className="item no-0" name="0" onClick={this.eventClick}> 0 </button>
        <button type="button" className="item dot" name="." onClick={this.eventClick}> . </button>
        <button type="button" className="item equal" name="=" onClick={this.eventClick}> =  </button>

      </main>
    );
  }
}

export default Calculator;
