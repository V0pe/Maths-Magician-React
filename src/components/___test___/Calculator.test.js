import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import Calculator from '../Calculator';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(() => {
  cleanup();
});

test('result render 0 initailly', () => {
  const { getByTestId } = render(<Calculator />);
  const result = getByTestId('result');

  expect(result.textContent).toBe('0');
});

test('1234567890 button renders with 1 in result area', () => {
  const { getByTestId } = render(<Calculator />);
  const no1 = getByTestId('no-1');
  const result = getByTestId('result');
  const no2 = getByTestId('no-2');
  const no3 = getByTestId('no-3');
  const no4 = getByTestId('no-4');
  const no5 = getByTestId('no-5');
  const no6 = getByTestId('no-6');
  const no7 = getByTestId('no-7');
  const no8 = getByTestId('no-8');
  const no9 = getByTestId('no-9');
  const no0 = getByTestId('0');

  fireEvent.click(no1);
  fireEvent.click(no2);
  fireEvent.click(no3);
  fireEvent.click(no4);
  fireEvent.click(no5);
  fireEvent.click(no6);
  fireEvent.click(no7);
  fireEvent.click(no8);
  fireEvent.click(no9);
  fireEvent.click(no0);

  expect(result.textContent).toBe('1234567890');
});

test('+ button rendered inbetween number in result area', () => {
  const { getByTestId } = render(<Calculator />);
  const no1 = getByTestId('no-1');
  const result = getByTestId('result');
  const no2 = getByTestId('no-2');
  const no3 = getByTestId('no-3');
  const add = getByTestId('+');

  fireEvent.click(no1);
  fireEvent.click(no2);
  fireEvent.click(add);
  fireEvent.click(no3);

  expect(result.textContent).toBe('12+3');
});

test('last operation button rendered', () => {
  const { getByTestId } = render(<Calculator />);
  const no1 = getByTestId('no-1');
  const result = getByTestId('result');
  const substract = getByTestId('-');
  const divide = getByTestId('/');
  const multiply = getByTestId('X');

  fireEvent.click(no1);
  fireEvent.click(substract);
  fireEvent.click(divide);
  fireEvent.click(multiply);

  expect(result.textContent).toBe('1x');
});

test('AC button rendered with clean slate', () => {
  const { getByTestId } = render(<Calculator />);
  const no1 = getByTestId('no-1');
  const result = getByTestId('result');
  const substract = getByTestId('-');
  const divide = getByTestId('/');
  const percent = getByTestId('%');
  const AC = getByTestId('AC');

  fireEvent.click(no1);
  fireEvent.click(substract);
  fireEvent.click(divide);
  fireEvent.click(percent);
  fireEvent.click(AC);

  expect(result.textContent).toBe('');

  fireEvent.click(no1);
  fireEvent.click(substract);
  fireEvent.click(no1);

  expect(result.textContent).toBe('1-1');
});

test('match snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('1 divide 1 should be 1', () => {
    const { getByTestId } = render(<Calculator />);
    const no1 = getByTestId('no-1');
    const result = getByTestId('result');
    const substract = getByTestId('-');
    const divide = getByTestId('/');
    const equal = getByTestId('=');
    const AC = getByTestId('AC');
  
    fireEvent.click(no1);
    fireEvent.click(divide);
    fireEvent.click(no1);
    fireEvent.click(equal);
  
    expect(result.textContent).toBe('1');
  });