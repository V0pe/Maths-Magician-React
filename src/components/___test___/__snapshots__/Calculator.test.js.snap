// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<main>
  <div
    className="item result"
    data-testid="result"
  >
    0
  </div>
  <button
    className="item reset"
    data-testid="AC"
    name="AC"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="item order"
    data-testid="+/-"
    name="+/-"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="item percent"
    data-testid="%"
    name="%"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="item divide"
    data-testid="/"
    name="รท"
    onClick={[Function]}
    type="button"
  >
    /
  </button>
  <button
    className="item no-7"
    data-testid="no-7"
    name="7"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="item no-8"
    data-testid="no-8"
    name="8"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="item no-9"
    data-testid="no-9"
    name="9"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="item multiply"
    data-testid="X"
    name="x"
    onClick={[Function]}
    type="button"
  >
    X
  </button>
  <button
    className="item no-4"
    data-testid="no-4"
    name="4"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="item no-5"
    data-testid="no-5"
    name="5"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="item no-6"
    data-testid="no-6"
    name="6"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="item substract"
    data-testid="-"
    name="-"
    onClick={[Function]}
    type="button"
  >
     - 
  </button>
  <button
    className="item no-1"
    data-testid="no-1"
    name="1"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="item no-2"
    data-testid="no-2"
    name="2"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="item no-3"
    data-testid="no-3"
    name="3"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="item add"
    data-testid="+"
    name="+"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="item no-0"
    data-testid="0"
    name="0"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="item dot"
    data-testid="."
    name="."
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="item equal"
    data-testid="="
    name="="
    onClick={[Function]}
    type="button"
  >
    = 
  </button>
</main>
`;
